### ISSUE TITLE
a few questions about freeze-backbone,freeze-bn,and lr

### ISSUE THREAD

**Issue by 2696120622** [xuannianz/EfficientDet#19]
Thanks for your great works. After walking through your code, I have a few questions about freeze-backbone,freeze-bn,lr and anchor ...
Train
STEP1: python3 train.py --snapshot imagenet --phi {0, 1, 2, 3, 4, 5, 6} --gpu 0 --random-transform --compute-val-loss --freeze-backbone --batch-size 32 --steps 1000 pascal datasets/VOC2012 to start training. The init lr is 1e-3.
STEP2: python3 train.py --snapshot xxx.h5 --phi {0, 1, 2, 3, 4, 5, 6} --gpu 0 --random-transform --compute-val-loss --freeze-bn --batch-size 4 --steps 10000 pascal datasets/VOC2012 to start training when val mAP can not increase during STEP1. The init lr is 1e-4 and decays to 1e-5 when val mAP keeps dropping down.
The training has two steps.
1. freeze-backbone and freeze-bn are applied in step1 and step2 respectively, Why? 
2. How should I set the learning rate?
3. In bbox_transform, you compute network targets with anchors and gt_boxes in the same way as fizyr/keras-retinanet. However, the source code of EfficientDet does not yet make public. And, the paper EfficientDet has little descriptions about how to compute network targets. 
4. Can you give some idea about the way of preprocessing the image in preprocess_image(self, image)?
5. In focal(alpha=0.25, gamma=2.0), however, the paper EfficientDet employ focal loss with alpha=0.5 and gamma=1.5.
Thanks!

**Comment by xuannianz** [comment]
1. freeze-backbone is to avoid wrecking pretrained weights of EfficientNet when lr is large. BatchNormalize exhibits a significant degradation in performance when batch size is small.
2. Start with a large one, if diverge, downscale it.
4. That's how EfficientNet does.
5. I know this. Have not found which is better.

**Comment by 2696120622** [comment]
Thanks.
However, the learning rate is fixed to 1e-3 in train.py.  Should I add the --lr with "parser.add_argument('--lr',               help='Learning rate.', type=float, default=1e-5)"  in parse_args(args) by myself for setting the learning rate?
